nomos-blend/network/src/tests/edge_core.rs

```diff
@@ -1,121 +1,130 @@
-use core::task::{Context, Poll, Waker};
-use std::collections::VecDeque;
+use std::task::{Context, Poll};
 
 use futures::StreamExt as _;
 use libp2p::{
     core::{transport::PortUse, Endpoint},
     swarm::{
-        ConnectionDenied, ConnectionId, FromSwarm, NetworkBehaviour, NotifyHandler, SwarmEvent,
-        THandler, THandlerInEvent, THandlerOutEvent, ToSwarm,
+        ConnectionDenied, ConnectionId, FromSwarm, NetworkBehaviour, SwarmEvent, THandler,
+        THandlerInEvent, THandlerOutEvent, ToSwarm,
     },
     Multiaddr, PeerId, Swarm, SwarmBuilder,
 };
+use nomos_blend_scheduling::membership::Membership;
+use rand::rngs::OsRng;
 
-use crate::edge::handler::{EdgeToCoreBlendConnectionHandler, FromBehaviour};
+use crate::edge::{self, handler::ToBehaviour, Error};
 
-pub(super) struct TestEdgeSenderBehaviour {
-    events_from_handler: VecDeque<THandlerOutEvent<Self>>,
-    events_to_handler: VecDeque<THandlerInEvent<Self>>,
-    waker: Option<Waker>,
-    peer_details: Option<(PeerId, ConnectionId)>,
+pub async fn edge_sender_swarm() -> (Swarm<edge::Behaviour<OsRng>>, Multiaddr) {
+    swarm(edge::Behaviour::new(None, OsRng)).await
 }
 
-impl TestEdgeSenderBehaviour {
-    pub(super) fn new() -> Self {
-        Self {
-            events_from_handler: VecDeque::new(),
-            events_to_handler: VecDeque::new(),
-            waker: None,
-            peer_details: None,
+pub async fn no_message_edge_sender_swarm() -> (Swarm<NoMessageSendBehaviour>, Multiaddr) {
+    swarm(NoMessageSendBehaviour {
+        inner: edge::Behaviour::new(None, OsRng),
+    })
+    .await
+}
+
+async fn swarm<Behaviour>(behaviour: Behaviour) -> (Swarm<Behaviour>, Multiaddr)
+where
+    Behaviour: NetworkBehaviour,
+{
+    let mut swarm = SwarmBuilder::with_new_identity()
+        .with_tokio()
+        .with_quic()
+        .with_behaviour(|_| behaviour)
+        .unwrap()
+        .build();
+    let _ = swarm
+        .listen_on("/ip4/127.0.0.1/udp/0/quic-v1".parse().unwrap())
+        .unwrap();
+    let listening_addr = wait_for_listening_address(&mut swarm).await;
+    (swarm, listening_addr)
+}
+
+async fn wait_for_listening_address<Behaviour>(swarm: &mut Swarm<Behaviour>) -> Multiaddr
+where
+    Behaviour: NetworkBehaviour,
+{
+    loop {
+        if let Some(SwarmEvent::NewListenAddr { address, .. }) = swarm.next().await {
+            break address;
         }
     }
+}
+
+pub struct NoMessageSendBehaviour {
```

Comment:
Can you add a comment on what this test behaviour does, and what it allows to test?